app/src/main/java/ar/com/wolox/android/login/LoginPresenter.kt

```diff
@@ -1,6 +1,37 @@
 package ar.com.wolox.android.login
 
+import ar.com.wolox.android.example.utils.UserSession
+import ar.com.wolox.android.login.utils.Errors
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import javax.inject.Inject
 
-class LoginPresenter @Inject constructor() : BasePresenter<LoginView>()
\ No newline at end of file
+class LoginPresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<LoginView>() {
+
+    private fun isEmailValid(email: String): Boolean {
+
+        return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
+    }
+
+    fun onLogInButtonClick(email: String, password: String) {
+
+        val totalErrors: MutableList<Errors> = ArrayList()
+
+        if (email.isEmpty()) totalErrors.add(Errors.EMPTY_EMAIL)
+
+        else if (!isEmailValid(email)) totalErrors.add(Errors.INVALID_EMAIL)
+
+        if (password.isEmpty()) totalErrors.add(Errors.EMPTY_PASSWORD)
+
+        if (totalErrors.isEmpty()) {
+
+            userSession.email = email
+            userSession.password = password
+        } else {
+
+            for (actualError in totalErrors) {
+
+                actualError.callAction(view!!)
+            }
+        }
+    }
```

Comment:
We usually prefer to have curly brackets in the if/else statements, even if they are a single line. Like:
```
if (email.isEmpty()) {
    totalErrors.add(Errors.EMPTY_EMAIL)
}
```